tests/ecdhp256-test.c

```diff
@@ -89,8 +93,134 @@ int main()
 	ok = ok && compare_and_print(64, point_compressed, compressed7);
 
 
+	printf("%s\n", "Wycheproof tests ECDH: ");
+		
+	for (int i = 0 ; i< sizeof(w_vectors)/sizeof(ecdhp256_w_i); i++)
+	{
+		printf("ECDH Wycheproof Test %d ", i );
+
+		uint8_t* decompressedPoint = (uint8_t*) malloc (sizeof (uint8_t) * 64);
+		bool flagDecompress = Hacl_P256_decompression_not_compressed_form(w_vectors[i].publicKey, decompressedPoint);
+
+		uint64_t success = Hacl_P256_ecp256dh_r(result, decompressedPoint, w_vectors[i].privateKey);
+
+		uint64_t fDU = 0;
+		if (flagDecompress = true)
+			fDU = success;
+		
+		if (w_vectors[i].flag != 1) 
+		{
+
+			ok = ok && ((success & fDU) == w_vectors[i].flag);
```

Comment:
the documentation states that `success` is non-zero if an error has occured

you now seem to be doing some bit-wise logic on the return value `success` -- are there some error codes that are encoded in the return value? what are the possible values for `w_vectors[i].flag`?